tools/crm_node.c

```diff
@@ -554,13 +554,21 @@ purge_node_from_cib(const char *node_name, long node_id)
         goto done;
     }
 
+    // Start a transaction
+    rc = cib->cmds->init_transaction(cib);
+    if (rc != pcmk_ok) {
+        rc = pcmk_legacy2rc(rc);
+        goto done;
+    }
+
     // Remove from configuration section
     xml = create_xml_node(NULL, XML_CIB_TAG_NODE);
     crm_xml_add(xml, XML_ATTR_UNAME, node_name);
     if (node_id > 0) {
         crm_xml_set_id(xml, "%ld", node_id);
     }
-    rc = cib->cmds->remove(cib, XML_CIB_TAG_NODES, xml, cib_sync_call);
+    rc = cib->cmds->remove(cib, XML_CIB_TAG_NODES, xml,
+                           cib_sync_call|cib_transaction);
```

Comment:
`cib_sync_call` no longer makes sense here and below. We won't process the request until we commit the transaction.